google/cloud/datastore/client.py

```diff
@@ -214,6 +228,10 @@ def _extended_lookup(
             **kwargs,
         )
 
+        # set new transaction id if we just started a transaction
+        if transaction and lookup_response.transaction:
```

Comment:
We will want this business logic to work for runQuery and runAggregation query Gapic requests as well. It looks like this code is strictly only for Gapic Lookup requests? If so then this business logic should be in place for runQuery and runAggregation Query as well.

Reply:
done